test/query.test.js

```diff
@@ -4349,4 +4349,38 @@ describe('Query', function() {
 
     assert.ok('title' in replacedDoc === false);
   });
+
+  it('handles $elemMatch with nested schema (gh-12902)', async function() {
```

Comment:
it looks like a new connection is opened and not closed in both of the following cases:
- when the test is successful
- when the test was not successful

re #12890

---

does this also work when just setting `strictQuery` on the query in question instead of requiring a whole new mongoose instance & connection?

Reply:
@hasezoey edited the suggestion to disconnect after the test, yes you can set `strictQuery` without a new connection, but wouldn't this affect other tests, the ones using default values for example

Reply:
This approach works, but it masks the underlying issue that, for some reason, global `strictQuery` is different in Deno than in Node. We had a couple of similar issues with our initial Deno work in #12397, don't have a good explanation for exactly why this issue is happening. We did some workarounds like https://github.com/Automattic/mongoose/pull/12397/commits/72124b88cc889918e904e4c769b751f382ad3df9 but don't have a good explanation.

Ended up just switching the #6658 tests to use `new mongoose.Mongoose()`, which is more correct anyway.